src/actions/dataExport.jsx

```diff
@@ -0,0 +1,60 @@
+// Tip of the hat to: 
+// https://stackoverflow.com/questions/35623656
+// https://stackoverflow.com/questions/3916191
+
+import { stateToQS } from './complaints'
+import { buildLink, simulateClick } from './domUtils'
+import { MODAL_HID, MODAL_SHOWN, MODAL_TYPE_DATA_EXPORT } from '../constants'
+
+const DATA_HOST = 'https://data.consumerfinance.gov'
+
+// ----------------------------------------------------------------------------
+// Action Creators
+
+export function showExportDialog() {
+  return {
+    type: MODAL_SHOWN,
+    modalType: MODAL_TYPE_DATA_EXPORT,
+    modalProps: {}
+  }
+}
+
+export function exportAllResults(format) {
+  return (dispatch, getState) => {
+    dispatch({type: MODAL_HID})
+
+    const uri = DATA_HOST + '/api/views/nsyy-je5y/rows.json'
```

Comment:
This may only export a smaller subset of results. I would recommend using the default view linked to from `complaint-database` to return the entire list.: 
`https://data.consumerfinance.gov/api/views/s6ew-h6mp/rows.json`